File: jobs/2026-06-11__09-29-21/shopify-source-hmac-verification__JJ8UF6t/artifacts/user/hookdeck-task/send_requests.py
import hmac
import hashlib
import base64
import json
import urllib.request
import os

secret = os.environ["SHOPIFY_WEBHOOK_SECRET"]
source_url = "https://hkdk.events/hoil433hzhilbl"

# Canonical JSON body - must use exact same bytes for both requests
body = json.dumps({"test": "shopify-verification", "timestamp": "2026-06-11"}, separators=(',', ':')).encode('utf-8')

# Compute correct HMAC SHA-256 signature
signature = base64.b64encode(
    hmac.new(secret.encode('utf-8'), body, hashlib.sha256).digest()
).decode('utf-8')

print(f"Body: {body}")
print(f"Correct signature: {signature}")

# Request 1: Correctly signed
print("\n--- Sending correctly signed request ---")
req1 = urllib.request.Request(source_url, data=body, method='POST')
req1.add_header('Content-Type', 'application/json')
req1.add_header('X-Shopify-Hmac-Sha256', signature)
try:
    resp1 = urllib.request.urlopen(req1)
    print(f"Response 1 status: {resp1.status}")
    print(f"Response 1 body: {resp1.read().decode()}")
except Exception as e:
    print(f"Response 1 error: {e}")

# Request 2: Tampered (wrong signature)
print("\n--- Sending tampered request ---")
req2 = urllib.request.Request(source_url, data=body, method='POST')
req2.add_header('Content-Type', 'application/json')
req2.add_header('X-Shopify-Hmac-Sha256', 'wrong_signature_value')
try:
    resp2 = urllib.request.urlopen(req2)
    print(f"Response 2 status: {resp2.status}")
    print(f"Response 2 body: {resp2.read().decode()}")
except Exception as e:
    print(f"Response 2 error: {e}")

print("\nDone sending requests.")

File: jobs/2026-06-11__08-14-38/shopify-source-hmac-verification__S6mHDSk/artifacts/user/hookdeck-task/send_requests.js
const crypto = require("crypto");

const SHOPIFY_WEBHOOK_SECRET = process.env.SHOPIFY_WEBHOOK_SECRET;
const SOURCE_URL = "https://hkdk.events/atuxgtvg9m8ntl";

if (!SHOPIFY_WEBHOOK_SECRET) {
  console.error("Missing SHOPIFY_WEBHOOK_SECRET environment variable!");
  process.exit(1);
}

const bodyObj = {
  id: 123456,
  email: "test@example.com",
  created_at: "2026-06-11T08:31:53Z",
  customer: {
    first_name: "John",
    last_name: "Doe"
  }
};

const rawBody = JSON.stringify(bodyObj);

// Compute HMAC SHA256 of rawBody
const hmac = crypto.createHmac("sha256", SHOPIFY_WEBHOOK_SECRET);
hmac.update(rawBody, "utf8");
const signature = hmac.digest("base64");

console.log("Raw Body:", rawBody);
console.log("Computed Signature (Base64):", signature);

async function sendRequest(isSigned) {
  const headers = {
    "Content-Type": "application/json",
  };

  if (isSigned) {
    headers["X-Shopify-Hmac-Sha256"] = signature;
  } else {
    // Send with an obviously wrong signature
    headers["X-Shopify-Hmac-Sha256"] = "obviously_wrong_signature_123456=";
  }

  console.log(`\nSending ${isSigned ? "SIGNED" : "UNSIGNED/TAMPERED"} request...`);
  try {
    const res = await fetch(SOURCE_URL, {
      method: "POST",
      headers: headers,
      body: rawBody,
    });

    console.log("Response Status:", res.status);
    const text = await res.text();
    console.log("Response Body:", text);
  } catch (err) {
    console.error("Error sending request:", err.message);
  }
}

async function run() {
  // Send correctly signed request
  await sendRequest(true);

  // Allow 2 seconds before sending the next
  await new Promise((resolve) => setTimeout(resolve, 2000));

  // Send tampered request
  await sendRequest(false);
}

run();
